src/views/Generator/AutoCorrelation/AutoCorrelation.tsx

```diff
@@ -0,0 +1,136 @@
+import { Box, Button, Flex, ScrollArea } from '@radix-ui/themes'
+import { Allotment } from 'allotment'
+import { useEffect, useState } from 'react'
+
+import { useListenProxyData } from '@/hooks/useListenProxyData'
+import { useGeneratorStore } from '@/store/generator'
+
+import { IntroductionMessage } from './IntroductionMessage'
+import { StatusPanel } from './StatusPanel'
+import { SuggestedRules } from './SuggestedRules'
+import { ValidationResults } from './ValidationResults'
+import { useGenerateRules } from './useGenerateRules'
+
+interface AutoCorrelationDialogProps {
+  close: () => void
+}
+
+export function AutoCorrelation({ close }: AutoCorrelationDialogProps) {
+  const rules = useGeneratorStore((state) => state.rules)
+  const saveRules = useGeneratorStore((state) => state.setRules)
+  const [checkedRuleIds, setCheckedRuleIds] = useState<string[]>([])
+
+  const { proxyData: validationRequests, resetProxyData: clearValidation } =
+    useListenProxyData()
+
+  const {
+    start,
+    suggestedRules,
+    isValid,
+    isLoading,
+    correlationStatus,
+    error,
+    messages,
+    outcomeReason,
+    stop,
+  } = useGenerateRules({
+    clearValidation: clearValidation,
+  })
+
+  // Abort streaming on unmount
+  useEffect(() => {
+    return () => {
+      window.studio.script.stopScript()
+      void stop()
+    }
+  }, [stop])
+
+  const handleAccept = () => {
+    const checkedRules = suggestedRules.filter((rule) =>
+      checkedRuleIds.includes(rule.id)
+    )
+    saveRules([...rules, ...checkedRules])
+    close()
+  }
+
+  const handleDiscard = () => {
+    close()
+  }
+
+  if (correlationStatus === 'not-started') {
+    return <IntroductionMessage onStart={start} />
+  }
+
+  return (
```

Comment:
I think the UI needs some work to better utilize the space and reduce duplication. I'll add concrete examples in this thread.

Reply:
Forgot to add one more comment about the dialog/its content: it doesn't work on all window sizes we support:

<img width="1312" height="774" alt="Image" src="https://github.com/user-attachments/assets/640c55bb-f3b3-481f-bca3-9f4e16f12f8e" />

I used the following hack in Settings some time ago:
```tsx
<Dialog.Content
  // This constraings could be anything. For example `1200px` and `750px` as it is in you core rn
  maxWidth="800px"
  maxHeight="640px"
  // This is the trick. It makes sure the dialog always fits on the screen 
  width="calc(100vw - 100px)"
  height="calc(100vh - 100px)"
>
```